### PR TITLE
Pin Nebari dependencies, set k8s version for GKE

### LINKED ISSUE #1623: [BUG] - `python-auth0` version `4.0.0` introduces breaking changes
### Describe the bug

Nebari can integrate with Auth0 and as such uses the `python-auth0` package. Two hours ago, Auth0 released a new version (4.0.0) which introduced a few breaking changes. ([See their migration guide for details](https://github.com/auth0/auth0-python/blob/master/V4_MIGRATION_GUIDE.md)). 

Now, when using the `nebari` CLI (with `python-auth0==4.0.0`), you will see the following error message:

```
Traceback (most recent call last):
  File "/Users/eskild/miniconda3/envs/nebari/bin/nebari", line 5, in <module>
    from nebari.cli.main import app
  File "/Users/eskild/miniconda3/envs/nebari/lib/python3.11/site-packages/nebari/cli/main.py", line 13, in <module>
    from nebari.cli.init import (
  File "/Users/eskild/miniconda3/envs/nebari/lib/python3.11/site-packages/nebari/cli/init.py", line 9, in <module>
    from nebari.initialize import render_config
  File "/Users/eskild/miniconda3/envs/nebari/lib/python3.11/site-packages/nebari/initialize.py", line 13, in <module>
    from nebari.provider.oauth.auth0 import create_client
  File "/Users/eskild/miniconda3/envs/nebari/lib/python3.11/site-packages/nebari/provider/oauth/auth0.py", line 4, in <module>
    from auth0.v3.authentication import GetToken
ModuleNotFoundError: No module named 'auth0.v3'
```

### Expected behavior

We expect the CLI to work.

### OS and architecture in which you are running Nebari

MacOS - all versions of Nebari

### How to Reproduce the problem?

Try `pip` installing `nebari` and then try and use the CLI.

### Anything else?

Since we don't pin `python-auth0`, all past versions of Nebari that are now installed, will run into this issue. This is even more troublesome given that our GitOps workflows (linting and deploying) pip install nebari each time they run and therefore will no longer work.

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

Fixes #1623

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

**Comment by iameskild** [comment]
I also removed `environment-dev.yaml`, I couldn't find anywhere in the Nebari org (including the integration test repo) it's used.

**Comment by viniciusdc** [comment]
@iameskild can you run IT on this branch just to make sure its deployable? I don't see any problems with just pinning (I do suppose we pinned with the current versions we were already using in the recent deployments) so this looks good to go

**Review by viniciusdc — APPROVED** [review]
LGTM

File: nebari/stages/input_vars.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ def stage_02_infrastructure(stage_outputs, config):
             "environment": config["namespace"],
             "region": config["google_cloud_platform"]["region"],
             "project_id": config["google_cloud_platform"]["project"],
+            "kubernetes_version": config["google_cloud_platform"]["kubernetes_version"],
```
Comment (viniciusdc) [inline nebari/stages/input_vars.py:67]: I was going to point this out, good catch. Are there any other inputs missing for the different cloud providers?

Comment (iameskild) [reply nebari/stages/input_vars.py:67]: I haven't noticed any other ones missing but I will keep my eye out for them. I issue I am currently running into is how to set the kubernetes version from the [Google Terraform provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster) 😅 I thought I could set the `master_version` but turns out that is a read-only attribute. Still investigating.

File: pyproject.toml [hunk 2]
```diff
@@ -51,21 +51,20 @@ classifiers = [
...
 
 dependencies = [
-    "auth0-python",
+    "auth0-python==4.0.0",
     "azure-identity==1.6.1",
     "azure-mgmt-containerservice==16.2.0",
-    "bcrypt",
-    "boto3",
-    "cloudflare",
-    "importlib_metadata; python_version<'3.8'",
```
Comment (costrouc) [inline pyproject.toml:60]: glad to see ununsed dependencies removed.

**Review by costrouc — APPROVED** [review]
Looks great to me. Glad we have kubernetes pinned on gcp now

**Pull request merged**